validator/sawtooth_validator/server/cli.py

```diff
@@ -194,17 +200,22 @@ def main(args=sys.argv[1:]):
                            human_readable_name='Log'):
         init_errors = True
 
+    if validator_config.public_uri is None:
```

Comment:
Let's get rid of the need for specifying public_uri in the common case -- there is no reason we have to specify this unless we are binding to multiple interfaces via 0.0.0.0 or behind NAT.  If network_endpoint is set to tcp://0.0.0.0:port and public_uri is not been set, then we should exit.  In all other cases, public_uri should default to network_endpoint.